common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -167,37 +168,42 @@ def course_start_date_is_default(start, advertised_start):
     return advertised_start is None and start == DEFAULT_START_DATE
 
 
-def _datetime_to_string(date_time, format_string, strftime_localized):
+def _datetime_to_string(date_time, format_string, time_zone, strftime_localized):
     """
     Formats the given datetime with the given function and format string.
 
-    Adds UTC to the resulting string if the format is DATE_TIME or TIME.
+    Adds time zone abbreviation to the resulting string if the format is DATE_TIME or TIME.
 
     Arguments:
         date_time (datetime): the datetime to be formatted
         format_string (str): the date format type, as passed to strftime
+        time_zone (pytz time zone): the time zone to convert to
         strftime_localized ((datetime, str) -> str): a nm localized string
             formatting function
     """
-    # TODO: Is manually appending UTC really the right thing to do here? What if date_time isn't UTC?
-    result = strftime_localized(date_time, format_string)
+    result = strftime_localized(date_time.astimezone(time_zone), format_string)
+    abbr = get_formatted_time_zone(time_zone, **{'abbr': True})
```

Comment:
This line of code is beautiful, compared to the one-liner with `TODO` from before 😃